framework/src/problems/FEProblemBase.C

```diff
@@ -845,7 +845,7 @@ FEProblemBase::initialSetup()
     {
       TIME_SECTION("computingInitialStatefulProps", 3, "Computing Initial Material Values");
 
-      ConstElemRange & elem_range = *_mesh.getActiveLocalElementRange();
+      ConstElemRange & elem_range = getEvaluableElementRange();
```

Comment:
I grabbed this change from @fdkong 's branch which is fixing the issue mentioned in #12036. This is required for stateful properties on ghosted elements to be properly evaluated.

I think the additional computational cost is small. But if it is a concern, I can add a flag in FEProblem to only optionally evaluate nonlocal elements.